Add release notes for Chocolatey Agent 0.13.0 (#392)
chocolatey/docs: input/en-us/agent/release-notes.md

* Please see https://github.com/chocolatey/chocolatey-licensed-issues/labels/AgentService
* Some issues may be held internally, please follow your support routes to learn more.

## 0.13.0 (February 28, 2022)

> :warning: **WARNING**
>
> The dependencies of the chocolatey-agent package have changed in this release, and it now requires Chocolatey Licensed Extension v3.2.0+.

### Breaking Changes

> - Central Management - Use new WCF interfaces/contracts when reporting information into CCM

Reviewer: What's a 'WCF interfaces / contract'? Remember these are written for the end users.

Can we also say 'Chocolatey Central Management' and not CCM or Central Management? I haven't changed 'issues' in the Improvements  section because they are issue titles.

Author: Happy to take suggestions on how else to say this.  WCF is Windows Communication Foundation, and it is how Chocolatey Agent communicates with Chocolatey Central Management.

Reviewer: I know we both know what they are. But that is a technical description of a change. If you had to explain that to somebody who doesn't know, what would we say?

'Enhanced communication method for reporting information in Chocolatey Central Management'?

Reviewer: As this doesn't requires a new version of Chocolatey Central Management, is it a breaking change?

Author: Yes, according to the new procedure that I followed:

> Changes to dependencies or other prerequisites should be noted in a warning at the top of the release notes and is regarded as a BREAKING CHANGE.

Reviewer: What dependencies are changing?

Author: Chocolatey Agent has a dependency on Chocolatey Licensed Extension where the new enhanced communication mechanism is defined, and it needs to use the latest 3.2.0 version.

>   - **NOTE:** There is no hard requirement to update to CCM 0.8.0 as this version of Chocolatey Agent will work in a backwards compatible way, but we do recommend updating to the latest version of CCM

Reviewer: ```suggestion
  - **NOTE:** This version of Chocolatey Agent is backwards compatible with versions of Chocolatey Central Management earlier than 0.8.0. However, we always recommend updating to the latest version of Chocolatey Central Management.
```

### Improvements

> - Central Management

Reviewer: ```suggestion
- Chocolatey Central Management
```

>   - Add logic to retry reporting of a deployment step result to CCM if it initially fails

Reviewer: ```suggestion
  - Add logic to retry reporting of a deployment step result to CCM if it initially fails.
```

>   - Ensure that no expanded sensitive variables are captured in log that is returned to CCM

Reviewer: ```suggestion
  - Ensure that no expanded sensitive variables are captured in log that is returned to CCM.
```
